kafka-client/src/main/java/dev/responsive/db/CassandraClient.java

```diff
@@ -43,7 +46,9 @@
  */
 public class CassandraClient {
 
+  private final ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(2);
```

Comment:
why keep the thread pool isolated to the cassandra client? It's probably not a big deal, but since we're embedding ourselves into customers' apps we should probably be a little conservative about creating thread pools.

Reply:
We don't use it anywhere else -- at the moment it's all internal to the CassandraClient. I moved it here as part of consolidating the schema creation/prep in each store to the CassandraClient which is where/why the executor is needed. Seemed silly to pass it around through the configs and extract it to the StateStores just to ultimately end up only needing it for the CassandraClient, then we might as well just move it there.

As an aside, we do create separate executors for the EndOffsetsPoller, but that's different.

Reply:
Going to go ahead and merge this once the tests pass, but if you're worried about this I can do a quick patch to move it back

That said, I'm envisioning that if we do end up needing an executor outside the CassandraClient class, we can/should still just access it via a standard getter on CassandraClient so that it's still clear who "owns" the executor and its lifecycle/scope. But like I said, if you're concerned we might accidentally forget about the executor here and end up creating more and blowing up the thread count, I'm happy to do a short followup